### PR TITLE
platform(secrets): modify checkov secrets scanner to scan all files based on ff

### PR DESCRIPTION
**Pull request by lirshindalman**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

### Description
add ff for enable_secret_scan_all_files and blacklist (black_list_secret_scan_lower) instead of whitelist (secrets_scan_file_type)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/main.py [hunk 1]
```diff
@@ -563,11 +565,10 @@ def add_parser_args(parser: ArgumentParser) -> None:
                env_var='CKV_SECRETS_SCAN_FILE_TYPE',
                action='append',
                help='add scan secret for requested files. You can specify this argument multiple times to add '
-                    'multiple file types. To scan all types (".tf", ".hcl", ".yml", ".yaml", ".json", ".template", '
-                    '".bicep", ".py", ".js", ".properties", ".pem", ".php", ".xml", ".ts", ".env", "Dockerfile", '
```
Comment (Eliran-Turgeman) [inline checkov/main.py:567]: @nimrodkor I think it's reverting your earlier fix for hcl and bicep from https://github.com/bridgecrewio/checkov/pull/3472/files

Comment (nimrodkor) [reply checkov/main.py:567]: But why?

File: checkov/secrets/runner.py [hunk 2]
```diff
@@ -152,23 +152,19 @@ def run(
             files_to_scan = files or []
             excluded_paths = (runner_filter.excluded_paths or []) + ignored_directories + [DEFAULT_EXTERNAL_MODULES_DIR]
             if root_folder:
-                secrets_scan_file_type = runner_filter.secrets_scan_file_type
-                if secrets_scan_file_type:
-                    secrets_scan_file_type_lower = [file_type.lower() for file_type in secrets_scan_file_type]
+                enable_secret_scan_all_files = runner_filter.enable_secret_scan_all_files
+                black_list_secret_scan = runner_filter.black_list_secret_scan
+                black_list_secret_scan_lower = [file_type.lower() for file_type in black_list_secret_scan] if black_list_secret_scan else []
```
Comment (Eliran-Turgeman) [inline checkov/secrets/runner.py:157]: If you use `black_list_secret_scan = runner_filter.black_list_secret_scan or []` instead,
you can remove the if statement on the next line 😄

File: checkov/main.py (hunk 1 above)
Comment (lirshindalman) [reply checkov/main.py]: fixed

File: checkov/secrets/runner.py (hunk 2 above)
Comment (lirshindalman) [reply checkov/secrets/runner.py:157]: thanks!

**Review by nimrodkor — APPROVED** [review]

File: checkov/runner_filter.py [hunk 3]
```diff
@@ -245,9 +249,13 @@ def from_dict(obj: Dict[str, Any]) -> RunnerFilter:
         run_image_referencer = obj.get('run_image_referencer')
         if run_image_referencer is None:
             run_image_referencer = False
+        enable_secret_scan_all_files = obj.get('enable_secret_scan_all_files')
+        if enable_secret_scan_all_files is None:
```
[the remark this answers is not in the document]
Comment (lirshindalman) [reply checkov/runner_filter.py]: only for typing - change to `bool(obj.get('enable_secret_scan_all_files'))`

**Review by Eliran-Turgeman — APPROVED** [review]

**Pull request merged**